df.history.xml

```diff
@@ -3327,28 +3324,32 @@
         </compound>
     </struct-type>
 
+    <struct-type type-name='relationship_event'>
+        <static-array name='event' type-name='int32_t' count='1024' comment="not included in the main list"/>
+        <static-array name='relationship' type-name='vague_relationship_type' base-type='int16_t' count='1024'/>
```

Comment:
Just caught this - `base-type` isn't a valid attribute of `static-array` according to the schema @BenLubar added recently. For some reason this didn't cause the check to fail until this PR was merged, maybe because the PR was made before the schema was added. For reference: https://travis-ci.org/github/DFHack/df-structures/builds/670832696#L246

I'm not sure if the schema needs to be corrected, but I'm going to work around this by having an `<enum>` as a child of the static-array

Reply:
In this case it would be possible to just remove the base-type attribute, as the enum is defined with that type as its base. However, the schema probably should be updated, as I can't see any logical reason for why the attribute shouldn't be allowed in this particular case.

Reply:
> However, the schema probably should be updated, as I can't see any logical reason for why the attribute shouldn't be allowed in this particular case.

If the code generator does not read it, the validator should not accept it (I don't know what the code generator does).

Reply:
The code generator appears to not support this at the moment:
```xml
    <struct-type type-name='test_foo'>
        <static-array name='relationship16' type-name='vague_relationship_type' base-type='int16_t' count='1024'/>
        <static-array name='relationship32' type-name='vague_relationship_type' base-type='int32_t' count='1024'/>
    </struct-type>
```
gives:
```c++
namespace df {
  struct DFHACK_EXPORT test_foo {
    df::vague_relationship_type relationship16[1024];
    df::vague_relationship_type relationship32[1024];
    static struct_identity _identity;
  public:
    test_foo();
  };
}
```
We would have to modify both to support this.

The tricky part is that `static-array`s can also contain things that are *not* enums, e.g. plain integer types. While we can verify that the `base-type` of an `<enum>` is valid, determining whether the `type-name` of a `static-array` is an enum type at the schema level (and only then allowing `base-type`) is somewhat harder. We _could_ do it by generating a list of all enum types and passing that into a dynamically-generated schema somehow, but I think it's simpler to just use the strategy in https://github.com/DFHack/df-structures/commit/98cb6941f651dd27d3ccdc9792ef01cab398c7a5#diff-35ab45624fa78d4a8c071f3c0fee5d76R3328-R3331